OktaLogger/OktaLoggerDestination.swift

```diff
@@ -155,13 +155,15 @@ open class OktaLoggerDestinationBase: NSObject, OktaLoggerDestinationProtocol {
     }
 
     open func addDefaultProperties(_ defaultProperties: [AnyHashable: Any]) {
-        self.defaultProperties.merge(defaultProperties, uniquingKeysWith: { (_, last) in last })
-        self.defaultPropertiesDescription = Self.description(of: self._defaultProperties)
+        var updatedProperties = self.defaultProperties
+        updatedProperties.merge(defaultProperties, uniquingKeysWith: { (_, last) in last })
+        self.defaultProperties = updatedProperties
     }
 
     open func removeDefaultProperties(for key: AnyHashable) {
-        defaultProperties.removeValue(forKey: key)
-        self.defaultPropertiesDescription = Self.description(of: self._defaultProperties)
+        var updatedProperties = defaultProperties
+        updatedProperties.removeValue(forKey: key)
+        defaultProperties = updatedProperties
```

Comment:
I was getting a crash on the description line. We only execute the lock on get/set and accessing the `self._defaultProperties` directly is unsafe. 

Changed to use the getter/setter locks which also update the description. Verified in debugging that `updatedProperties` object is a copy not a reference to `self.defaultProperties`